legend-pure-core/legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/compiler/postprocessing/processor/valuespecification/FunctionExpressionProcessor.java

```diff
@@ -616,6 +616,10 @@ private static boolean processLambda(FunctionExpression functionExpression, Proc
                                 TypeInferenceContext typeInferenceContext = state.getTypeInferenceContext();
                                 typeInferenceContext.register(templateReturnType, concreteGenericType, typeInferenceContext.getParent(), observer);
                             }
+                            else if (concreteGenericType != null)// type arguments might not be concreate
```

Comment:
If the template is concreate, it might still have type arguments that are not.  

If we got a concreate type, recursively go thru the type arguments to capture any type that can be properly inferred.